packages/blade/src/components/Input/BaseInput/BaseInput.tsx

```diff
@@ -77,26 +170,60 @@ export const BaseInput = ({
   type = 'text',
   defaultValue,
   name,
+  value,
   onChange,
+  isDisabled,
+  neccessityIndicator,
+  validationState,
+  errorText,
+  helpText,
+  successText,
+  isRequired,
 }: BaseInputProps): ReactElement => {
-  console.log({
-    label,
-    labelPosition,
-  });
-
-  const handleOnChange: HandleOnChange = ({ event, inputName, value }) => {
-    console.log({ event, name: inputName, value: event?.target.value });
-    onChange?.({ event, name: inputName, value });
-  };
+  const { handleOnChange } = useInput({ defaultValue, value, onChange });
+  const { labelId, inputId, helpTextId, errorTextId, successTextId } = useFormId('input-field');
 
   return (
-    <StyledBaseInput
-      name={name}
-      type={type}
-      defaultValue={defaultValue}
-      placeholder={placeholder}
-      // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-      onChange={(event) => handleOnChange({ event, inputName: name, value: '' })}
-    />
+    <>
+      <Box
+        display="flex"
+        flexDirection={labelPosition === 'left' ? 'row' : 'column'}
+        justifyContent={labelPosition === 'left' ? 'center' : 'normal'}
+        alignItems={labelPosition === 'left' ? 'center' : 'normal'}
+      >
+        <FormLabel
+          as="label"
+          neccessityIndicator={neccessityIndicator}
+          id={labelId}
```

Comment:
Oh actually ID isn't needed when using it `as=label` can you remove it and make the ID prop on FormLabel optional.

Reply:
it'll be needed right? because we are not wrapping the text input inside label so we need to associate the label with the input id

Reply:
label doesn't need ID. it only needs htmlFor. 

input needs ID.

Reply:
okay i confused these 2 props.